### DISCUSSION TITLE
PHP 8.1

### DISCUSSION THREAD

**Discussion by mojowill** [spatie/laravel-livewire-wizard#89] [Q&A]
Your composer file for this repo states PHP 8.1+ however a dependant package of yours https://github.com/spatie/laravel-collection-macros/commit/f99686ac11af7aaf020d7b4e4bb75402a0029fb9 has dropped support for anything but 8.2 for some reason?

I'm running 8.1 on Vapor and now this won't work for us?

**Comment by ju5t** [comment]
Are you not able to pin the macro package to v7.12.5, which is the latest before 8.2 was dropped? As this package depends on 7.11 that should be fine.

**Reply by mojowill** [reply]
Seems like a hacky workaround, both are Spatie packages so the expectation is that if A says it supports PHP 8.1 it should?

**Reply by ju5t** [reply]
It does support 8.1?

I just tried to install it on a fresh installation of Laravel and it installed v7.12.5 of the macro's package automatically. If this isn't happening for you, the macro package might be locked at a higher version on your app.

```bash
➜  composer require spatie/laravel-livewire-wizard
./composer.json has been updated
Running composer update spatie/laravel-livewire-wizard
Loading composer repositories with package information
Updating dependencies
Lock file operations: 4 installs, 0 updates, 0 removals
  - Locking livewire/livewire (v2.12.3)
  - Locking spatie/laravel-collection-macros (7.12.5)
  - Locking spatie/laravel-livewire-wizard (1.2.0)
  - Locking spatie/laravel-package-tools (1.15.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 4 installs, 0 updates, 0 removals
  - Downloading spatie/laravel-collection-macros (7.12.5)
  - Downloading spatie/laravel-livewire-wizard (1.2.0)
  - Installing spatie/laravel-package-tools (1.15.0): Extracting archive
  - Installing spatie/laravel-collection-macros (7.12.5): Extracting archive
  - Installing livewire/livewire (v2.12.3): Extracting archive
  - Installing spatie/laravel-livewire-wizard (1.2.0): Extracting archive
```

**Comment by mojowill** [comment]
Pinning 7.12.5 works for me. My point is I shouldn’t have to. I’d expect
some hoops to jump through if this was a child dependant from a third party
but it’s not, it’s another Spatie package.

On Wed, 26 Jul 2023 at 20:38, ju5t ***@***.***> wrote:

> It does support 8.1?
>
> I just tried to install it on a fresh installation of Laravel and it
> installed v7.12.5 of the macro's package automatically. If this isn't
> happening for you, the macro package might be locked at a higher version on
> your app.
>
> ➜  composer require spatie/laravel-livewire-wizard
> ./composer.json has been updated
> Running composer update spatie/laravel-livewire-wizard
> Loading composer repositories with package information
> Updating dependencies
> Lock file operations: 4 installs, 0 updates, 0 removals
>   - Locking livewire/livewire (v2.12.3)
>   - Locking spatie/laravel-collection-macros (7.12.5)
>   - Locking spatie/laravel-livewire-wizard (1.2.0)
>   - Locking spatie/laravel-package-tools (1.15.0)
> Writing lock file
> Installing dependencies from lock file (including require-dev)
> Package operations: 4 installs, 0 updates, 0 removals
>   - Downloading spatie/laravel-collection-macros (7.12.5)
>   - Downloading spatie/laravel-livewire-wizard (1.2.0)
>   - Installing spatie/laravel-package-tools (1.15.0): Extracting archive
>   - Installing spatie/laravel-collection-macros (7.12.5): Extracting archive
>   - Installing livewire/livewire (v2.12.3): Extracting archive
>   - Installing spatie/laravel-livewire-wizard (1.2.0): Extracting archive
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spatie/laravel-livewire-wizard/discussions/89#discussioncomment-6555139>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGTKRVJHRCN53JK5KAYENTXSFW4LANCNFSM6AAAAAA2YWLMPQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <spatie/laravel-livewire-wizard/repo-discussions/89/comments/6555139@
> github.com>
>
-- 

Will Wilson

Chief Product and Technology Officer


Think Digital Group | 30 Cannon Street | London | EC4M 6XH

Mobile: +44 (0)7799 194 394 | DDI: +44 (0)1582 349 700

Website: www.thinkdigitalgroup.com


[image: Think Digital Group Logo] <https://www.thinkdigitalgroup.com/>

Asset TV <https://www.asset.tv/> | Insure TV <https://www.insuretv.net/>

Book a meeting with me
<https://calendar.google.com/calendar/u/0/appointments/schedules/AcZssZ2KIgJycdc6wvAjCo2r1rcjr4jU2JxwFlPtwxiiEyV_NH00KGx5rpiQooPutxmy6EnaLFm-czuD>

**Reply by ju5t** [reply]
I don't know why you have to, because you're right, you shouldn't. But you're not right about this being a Spatie issue. As I said, I tested it on a fresh installation and I didn't have to jump through any hoops with 8.1. It immediately picked up the correct version of the macro package, which is compatible with PHP 8.1.
